### PR TITLE
Make next as dependency of `@next/third-parties` package

### PR DESCRIPTION
**Pull request by huozhi**
ensure `next` is the dependency of third-parties package so turbo will build next first then `@next/third-parties` to avoid type failing

### CODE REVIEW

**Comment by ijjk** [comment]
## Failing test suites 
Commit: e6daffd432213c4786704516bbbd1bfbd2d0a015

`pnpm testheadless test/e2e/app-dir/third-parties/basic.test.ts` 
- @next/third-parties basic usage > renders YoutubeEmbed
- @next/third-parties basic usage > renders GoogleMapsEmbed

<details>
<summary>Expand output</summary>

  ● @next/third-parties basic usage › renders YoutubeEmbed

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: 0

      15 |       const baseContainer = $('[data-ntpc="YoutubeEmbed"]')
      16 |       const youtubeContainer = $('lite-youtube')
    > 17 |       expect(baseContainer.length).toBe(1)
         |                                    ^
      18 |       expect(youtubeContainer.length).toBe(1)
      19 |     })
      20 |

      at Object.<anonymous> (e2e/app-dir/third-parties/basic.test.ts:17:36)

  ● @next/third-parties basic usage › renders GoogleMapsEmbed

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: 0

      26 |         '[src="https://www.google.com/maps/embed/v1/place?key=XYZ&q=Brooklyn+Bridge,New+York,NY"]'
      27 |       )
    > 28 |       expect(baseContainer.length).toBe(1)
         |                                    ^
      29 |       expect(mapContainer.length).toBe(1)
      30 |     })
      31 |   }

      at Object.<anonymous> (e2e/app-dir/third-parties/basic.test.ts:28:36)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

File: packages/third-parties/package.json [hunk 1]
```diff
@@ -16,19 +16,19 @@
   "scripts": {
     "build": "rm -rf dist && tsc -d -p tsconfig.json",
     "prepublishOnly": "cd ../../ && turbo run build",
-    "updateThirdParties": "rm -rf src/**/index.tsx && node scripts/update-third-parties",
+    "update-third-parties": "rm -rf src/**/index.tsx && node scripts/update-third-parties",
     "dev": "tsc -d -w -p tsconfig.json",
     "typescript": "tsec --noEmit -p tsconfig.json"
   },
   "dependencies": {
+    "next": "^13.0.0",
```
[the remark this answers is not in the document]
Comment (huozhi) [reply packages/third-parties/package.json:24]: This package also uses `"next/script"`, previously we specify it as peer dependency but I think it make sense to make it as dependencies as we really used it.

**Review by styfle — DISMISSED** [review]

**Review by shuding — DISMISSED** [review]

File: packages/third-parties/package.json (hunk 1 above)
Comment (ijjk) [inline packages/third-parties/package.json:24]: Shouldn't this be a `devDependency` as well as a `peerDependency`?

File: packages/third-parties/package.json [hunk 2]
```diff
@@ -16,19 +16,19 @@
...
     "third-party-capital": "1.0.17"
   },
   "devDependencies": {
     "outdent": "0.8.0",
```
Comment (ijjk) [inline packages/third-parties/package.json:28]:
```suggestion
    "next": "13.4.14",
    "outdent": "0.8.0",
```

File: packages/third-parties/package.json [hunk 3]
```diff
@@ -16,19 +16,19 @@
...
     "prettier": "2.5.1"
   },
   "peerDependencies": {
-    "next": "^13.0.0",
     "react": "^18.2.0"
```
Comment (ijjk) [inline packages/third-parties/package.json:32]:
```suggestion
    "next": "^13.0.0",
    "react": "^18.2.0"
```

File: packages/third-parties/package.json (hunk 2 above)
Comment (DuCanhGH) [reply packages/third-parties/package.json:28]: I think `workspace:*` will be a better option than `13.4.14` as we won't have to manually replace the value whenever a new version of `next` is released.

**Review by styfle — DISMISSED** [review]

File: packages/third-parties/package.json (hunk 2 above)
Comment (ijjk) [reply packages/third-parties/package.json:28]: Lerna takes care of this for us

**Review by ijjk — APPROVED** [review]

**Pull request merged**